istioctl/cmd/proxyconfig.go

```diff
@@ -116,13 +118,21 @@ var stringToLevel = map[string]Level{
 }
 
 var (
-	loggerLevelString = ""
-	reset             = false
+	loggerLevelString      = ""
+	reset                  = false
+	once                   sync.Once
+	isZtunnelPodKubeClient kube.CLIClient
 )
 
-var isZtunnelPod = func(podName, podNamespace string) bool {
-	client, _ := kubeClient(kubeconfig, configContext)
-	return ambientutil.IsZtunnelPod(client, podName, podNamespace)
+var isZtunnelPod = func(podName, podNamespace string) (bool, error) {
+	var err error
```

Comment:
From the second call to `isZtunnelPod`, the `err` variable will always be nil.

Reply:
Yes

Reply:
Previous version never checked for an error on kube client initialization. This addresses that. And yeah, it's true it only happens once but that's because of the nature of once.Do(). Error will always be nil after the first call but we still need to ensure we error check for that first run where initialization of the kube client happens.

Does that make sense? Is there a more elegant way of handling that @wulianglongrd ?

Reply:
I think that if you want to always hold the error of the first call, you can use a global variable, which is defined the same as `isZtunnelPodKubeClient`.

Reply:
Good catch! I think this makes sense and has the added benefit of bringing `isZtunnelPod` back to being a one value return. Here's a PR: https://github.com/istio/istio/pull/45160

@wulianglongrd